recipe/build.sh

```diff
@@ -1,15 +1,21 @@
 #!/bin/sh
 
-libtoolize --copy
-autoreconf -vfi
-
-./configure          \
-  --prefix=${PREFIX} \
-  --enable-static    \
-  --enable-shared    \
-  --without-octave   \
-  --without-matlab   \
-  --without-guile
-
-make -j${CPU_COUNT} ${VERBOSE_AT}
-make install ${VERBOSE_AT}
+if [ "$(uname)" = "Darwin" ]; then
+  # osx uses a temp directory which messes up with cmake finding libraries.
+  cd "${SRC_DIR}" || exit
+fi
+
+mkdir build && cd build || exit
+cmake \
+  -DCMAKE_PREFIX_PATH=${PREFIX} \
+  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+  -DCMAKE_INSTALL_LIBDIR=lib \
+  -DNLOPT_GUILE=OFF -DNLOPT_MATLAB=OFF -DNLOPT_OCTAVE=OFF \
```

Comment:
Why do we disable guile, matlab and octave here but not for Windows?

Reply:
I would probably disable python. See my comment at https://github.com/AnacondaRecipes/nlopt-feedstock/pull/1#issuecomment-2062007729. And yeah, if we disable something on unix, I think we should also disable it on Windows. Explicit is better than implicit.

Reply:
That's fair. Fixed it.